From: Jeremy Bicha <jbicha@debian.org>
Date: Thu, 23 Sep 2021 18:54:14 -0400
Subject: Revert "xsettings: Add support for XFixes ClientDisconnectMode"

This reverts commit e10a1cc5e76b9ef48e3de88b9562c39e444ea8c4.

Debian doesn't have libxfixes 6.0.0 yet
https://bugs.debian.org/994957
---
 meson.build                           | 1 -
 plugins/xsettings/meson.build         | 1 -
 plugins/xsettings/xsettings-manager.c | 3 ---
 plugins/xsettings/xsettings-manager.h | 1 -
 4 files changed, 6 deletions(-)

diff --git a/meson.build b/meson.build
index 6bdd7d0..7c211fb 100644
--- a/meson.build
+++ b/meson.build
@@ -106,7 +106,6 @@ pango_dep = dependency('pango', version: '>= 1.20.0')
 polkit_gobject_dep = dependency('polkit-gobject-1', version: '>= 0.114')
 upower_glib_dep = dependency('upower-glib', version: '>= 0.99.8')
 x11_dep = dependency('x11')
-xfixes_dep = dependency('xfixes', version: '>= 6.0')
 
 enable_systemd = get_option('systemd')
 if enable_systemd
diff --git a/plugins/xsettings/meson.build b/plugins/xsettings/meson.build
index 49ed70f..9adb102 100644
--- a/plugins/xsettings/meson.build
+++ b/plugins/xsettings/meson.build
@@ -14,7 +14,6 @@ sources = gsd_xsettings_gtk + fc_monitor + wm_button_layout_translation + files(
 deps = plugins_deps + [
   gtk_dep,
   x11_dep,
-  xfixes_dep,
   libcommon_dep,
   gsettings_desktop_dep,
   dependency('fontconfig')
diff --git a/plugins/xsettings/xsettings-manager.c b/plugins/xsettings/xsettings-manager.c
index a154ede..947cc9e 100644
--- a/plugins/xsettings/xsettings-manager.c
+++ b/plugins/xsettings/xsettings-manager.c
@@ -131,9 +131,6 @@ xsettings_manager_new (Display                *display,
 
   char buffer[256];
 
-
-  XFixesSetClientDisconnectMode (display, XFixesClientDisconnectFlagTerminate);
-
   manager = g_slice_new (XSettingsManager);
 
   manager->display = display;
diff --git a/plugins/xsettings/xsettings-manager.h b/plugins/xsettings/xsettings-manager.h
index 6346b09..a8cf151 100644
--- a/plugins/xsettings/xsettings-manager.h
+++ b/plugins/xsettings/xsettings-manager.h
@@ -24,7 +24,6 @@
 #define XSETTINGS_MANAGER_H
 
 #include <X11/Xlib.h>
-#include <X11/extensions/Xfixes.h>
 #include "xsettings-common.h"
 
 typedef struct _XSettingsManager XSettingsManager;
